File: app/src/main/java/com/boardgamegeek/sorter/LocationsSorterFactory.java
package com.boardgamegeek.sorter;

import android.content.Context;
import android.support.annotation.NonNull;

public class LocationsSorterFactory {
	public static final int TYPE_NAME = 1;
	public static final int TYPE_QUANTITY = 2;
	public static final int TYPE_DEFAULT = TYPE_NAME;

	@NonNull
	public static LocationsSorter create(@NonNull Context context, int type) {
		switch (type) {
			case TYPE_QUANTITY:
				return new LocationsQuantitySorter(context);
			case TYPE_NAME:
			default:
				return new LocationsNameSorter(context);
		}
	}
}

File: app/src/main/java/com/boardgamegeek/sorter/LocationsSorterFactory.kt
package com.boardgamegeek.sorter

import android.content.Context

class LocationsSorterFactory {
    companion object {
        const val TYPE_NAME = 1
        const val TYPE_QUANTITY = 2
        const val TYPE_DEFAULT = TYPE_NAME

        @JvmStatic
        fun create(context: Context, type: Int): LocationsSorter {
            return when (type) {
                TYPE_QUANTITY -> LocationsQuantitySorter(context)
                TYPE_NAME -> LocationsNameSorter(context)
                else -> LocationsNameSorter(context)
            }
        }
    }
}
